bip-descriptors-non-segwit.mediawiki

```diff
@@ -0,0 +1,83 @@
+<pre>
+  BIP: bip-descriptors-non-segwit
+  Layer: Applications
+  Title: Non-segwit Output Script Descriptors
+  Author: Pieter Wuille <pieter@wuille.net>
+          Andrew Chow <andrew@achow101.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-descriptors-non-segwit
+  Status: Draft
+  Type: Informational
+  Created: 2021-06-27
+  License: BSD-2-Clause
+</pre>
+
+==Abstract==
+
+This document specifies <tt>pk()</tt>, <tt>pkh()</tt>, and <tt>sh()</tt> output script descriptors.
+<tt>pk()</tt> descriptors take a key and produces a P2PK output script.
+<tt>pkh()</tt> descriptors take a key and produces a P2PKH output script.
+<tt>sh()</tt> descriptors take a script and produces a P2SH output script.
+
+==Copyright==
+
+This BIP is licensed under the BSD 2-clause license.
+
+==Motivation==
+
+Prior to the activation of Segregated Witness, there were 3 main standard output script formats: P2PK, P2PKH, and P2SH.
+These expressions allow specifying those formats as a descriptor.
+
+==Specification==
+
+Three new script expressions are defined: <tt>pk()</tt>, <tt>pkh()</tt>, and <tt>sh()</tt>.
+
+===<tt>pk()</tt>===
+
+The <tt>pk(KEY)</tt> expression can be used in any context or level of a descriptor.
+It takes a single key expression as an argument and produces a P2PK output script.
+Depending on the higher level descriptors, there may be restrictions on the type of public keys that can be included.
+Such restrictions will be specified by those descriptors.
+
+The output script produced is:
+<pre>
+<KEY> OP_CHECKSIG
+</pre>
+
+===<tt>pkh()</tt>===
+
+The <tt>pkh(KEY)</tt> expression can be used as a top level expression, or inside of either a <tt>sh()</tt> or <tt>wsh()</tt> descriptor.
```

Comment:
Is there any special reason for the description of `pk()` script expression and `pkh()` script expression different? Also, we are using wsh without actually defining/(referring to) what it is.

Reply:
`pk` is allowed inside of `tr` but `pkh` is not.

> Also, we are using wsh without actually defining/(referring to) what it 

These definitions all have mentions of descriptors from other documents. I haven't really figured out a good way to do.

Reply:
> pk is allowed inside of tr but pkh is not.

Makes sense. 

> These definitions all have mentions of descriptors from other documents. I haven't really figured out a good way to do it.

I think the main descriptor BIP could be kept updated with all the top-level contexts and a very high-level summary. How about something like?

```
Legacy output descriptors: sh(), pk(), pkh(), multi(), sorted_multi()
Segwitv0 output Desctipors: wsh(), wpkh(), 
Taproot output Descriptor: tr(),
```

Reply:
I've added a listing of expressions as appendices.